sql/core/src/main/scala/org/apache/spark/sql/execution/local/HashJoinNode.scala

```diff
@@ -73,10 +60,21 @@ case class HashJoinNode(
     }
   }
 
+  /** Sets the HashedRelation used by this node. */
```

Comment:
I would add a sentence to say this needs happen before the first `next` is called.